pages/docs/reference/mpp-dsl-reference.md

```diff
@@ -0,0 +1,860 @@
+<div class="sample" markdown="1" theme="idea" mode='groovy'>
+
+```groovy
+kotlin { 
+    sourceSets { 
+        myMain { /* ... */ } // create or configure a source set by the name 'myMain' 
+    }
+}
+``` 
+
+</div>
+</div>
+
+<div class="multi-language-sample" data-lang="kotlin">
+<div class="sample" markdown="1" theme="idea" mode='kotlin' data-highlight-only>
+
+```kotlin
+kotlin { 
+    sourceSets { 
+        val myMain by creating { /* ... */ } // create a new source set by the name 'MyMain'
+    }
+}
+```
+
+</div>
+</div>
+
+Note that a newly created source set isn’t connected to other ones. To use it in the project’s compilations,
+connect it with other source sets as described in [Connecting source sets](building-mpp-with-gradle.html#connecting-source-sets).
+
+### Source set parameters
+
+Configurations of source sets are stored inside the corresponding blocks of `sourceSets`. A source set has the following parameters:
+
+|**Name**|**Description**| 
+| --- | --- |
+|`kotlin.srcDir`|Location of Kotlin source files inside the source set directory.|
+|`resources.srcDir`|Location of resources inside the source set directory.|
+|`dependsOn`|Connection with another source set. The instructions on connecting source sets are provided in [Connecting source sets](building-mpp-with-gradle.html#connecting-source-sets).|
+|`dependencies`|[Dependencies](#dependencies) of the source set.|
+|`languageSettings`|[Language settings](building-mpp-with-gradle.html#language-settings) applied to the source set.|
+
+<div class="multi-language-sample" data-lang="groovy">
+<div class="sample" markdown="1" theme="idea" mode='groovy'>
+
+```groovy
+kotlin { 
+    sourceSets { 
+        commonMain {
+            kotlin.srcDir('src')
+            resources.srcDir('res')
+            
+            dependencies {
+                /* ... */
+            }           
+        } 
+    }
+}
+``` 
+
+</div>
+</div>
+
+<div class="multi-language-sample" data-lang="kotlin">
+<div class="sample" markdown="1" theme="idea" mode='kotlin' data-highlight-only>
+
+```kotlin
+kotlin { 
+    sourceSets { 
+        val commonMain by getting {
+            kotlin.srcDir("src")
+            resources.srcDir("res")
+            
+            dependencies {
+                /* ... */
+            } 
+        } 
+    }
+}
+```
+
+</div>
+</div>
+
+## Compilations
+
+A target can have one or more compilations, for example, for production or testing. There are [predefined compilations](#predefined-compilations)
+that are added automatically upon target creation. Developers can additionally create [custom compilations](#custom-compilations).
+
+To refer to all or some particular compilations of a target, use the `compilations` object collection.
+From `compilations`, you can refer to a compilation by its name.
+
+### Predefined compilations
+
+Predefined compilations are created automatically for each target of a project except for Android targets.
```

Comment:
I think it makes sense to explain the gist of which compilation are created by default for projects with Android targets. @h0tk3y could you help with the right sentence, please?